### PR TITLE
Fix UNNECESSARY_TYPE_CHECK_TRUE

### PR DESCRIPTION
**Pull request by scheglov**
We would like to land https://dart-review.googlesource.com/c/sdk/+/190360, and need to fix existing violations.

### CODE REVIEW

**Review by ditman — APPROVED** [review]
LGTM. Let's wait for the tests to pass, though.

File: lib/web_ui/test/channel_buffers_test.dart [hunk 1]
```diff
@@ -23,7 +23,7 @@ void main() {
 
 ByteData _makeByteData(String str) {
   final Uint8List list = utf8.encode(str) as Uint8List;
-  final ByteBuffer buffer = list is Uint8List ? list.buffer : Uint8List.fromList(list).buffer;
```
Comment (ditman) [inline lib/web_ui/test/channel_buffers_test.dart:26]: /cc @yjbanov I wonder in what cases we needed this `is` check; is the utf8 encoder platform-dependant somehow?

Comment (yjbanov) [reply lib/web_ui/test/channel_buffers_test.dart:26]: /cc @Hixie who wrote this code and @mdebbar who LGTM'd it :)

Interestingly, why did this same check not catch the other file: https://github.com/flutter/engine/blob/master/testing/dart/channel_buffers_test.dart#L16

Comment (ditman) [reply lib/web_ui/test/channel_buffers_test.dart:26]: /cc @scheglov had you seen https://github.com/flutter/engine/blob/master/testing/dart/channel_buffers_test.dart#L16?

Comment (scheglov) [reply lib/web_ui/test/channel_buffers_test.dart:26]: Yes, I see it now, and one more place as well. 

The repo is not clean for me, there are many classes missing, there are probably some configuration step that I missed.

Comment (ditman) [reply lib/web_ui/test/channel_buffers_test.dart:26]: @scheglov there's quite a bunch of setup required for the engine repo, have you seen its [setup instructions](https://github.com/flutter/flutter/wiki/Setting-up-the-Engine-development-environment)?

**Comment by ditman** [comment]
```
00:09 +0: loading /b/s/w/ir/x/w/flutter/dev/automated_tests/test_smoke_test/pass_test.dart                                                                                                             
00:10 +0: loading /b/s/w/ir/x/w/flutter/dev/automated_tests/test_smoke_test/pass_test.dart                                                                                                             
00:10 +0 -1: loading /b/s/w/ir/x/w/flutter/dev/automated_tests/test_smoke_test/pass_test.dart [E]                                                                                                      
  Failed to load "/b/s/w/ir/x/w/flutter/dev/automated_tests/test_smoke_test/pass_test.dart": Compilation failed for testPath=/b/s/w/ir/x/w/flutter/dev/automated_tests/test_smoke_test/pass_test.dart

00:10 +0 -1: Some tests failed.                                                                                                                                                                        

../../packages/flutter_test/lib/src/frame_timing_summarizer.dart:41:50: Error: The getter 'layerCacheCount' isn't defined for the class 'FrameTiming'.
 - 'FrameTiming' is from 'dart:ui'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'layerCacheCount'.
      data.map<int>((FrameTiming datum) => datum.layerCacheCount),
                                                 ^^^^^^^^^^^^^^^
../../packages/flutter_test/lib/src/frame_timing_summarizer.dart:45:50: Error: The getter 'layerCacheBytes' isn't defined for the class 'FrameTiming'.
 - 'FrameTiming' is from 'dart:ui'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'layerCacheBytes'.
      data.map<int>((FrameTiming datum) => datum.layerCacheBytes),
                                                 ^^^^^^^^^^^^^^^
../../packages/flutter_test/lib/src/frame_timing_summarizer.dart:49:50: Error: The getter 'pictureCacheCount' isn't defined for the class 'FrameTiming'.
 - 'FrameTiming' is from 'dart:ui'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'pictureCacheCount'.
      data.map<int>((FrameTiming datum) => datum.pictureCacheCount),
                                                 ^^^^^^^^^^^^^^^^^
../../packages/flutter_test/lib/src/frame_timing_summarizer.dart:53:50: Error: The getter 'pictureCacheBytes' isn't defined for the class 'FrameTiming'.
 - 'FrameTiming' is from 'dart:ui'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'pictureCacheBytes'.
      data.map<int>((FrameTiming datum) => datum.pictureCacheBytes),
                                                 ^^^^^^^^^^^^^^^^^

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
ERROR: Last command exited with 1 (expected: zero).
Command: ../../bin/flutter test --test-randomize-ordering-seed=20210914 --local-engine /b/s/w/ir/x/w/recipe_cleanup/builder/src/out/host_debug_unopt test_smoke_test/pass_test.dart
Relative working directory: dev/automated_tests
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
```

Maybe this branch needs a rebase with `master`?

**Comment by ditman** [comment]
If the engine repo is giving you grief, @scheglov, I can get this merged tomorrow!

**Comment by scheglov** [comment]
I rebased the PR, so tests pass now.
So, I think I will land it.

**Pull request merged**